middleware/kubernetes/kubernetes.go

```diff
@@ -92,13 +101,27 @@ func (k *Kubernetes) Services(state request.Request, exact bool, opt middleware.
 		return nil, nil, e
 	}
 
+	if state.Name() == DefaultNSName+"."+r.zone && state.Type() == "A" {
```

Comment:
`DefaultNSName` could also include a `.`

Also: shall we move some of this logic on another file? This one getting big, `ns.go` or something?